lightning/src/util/config.rs

```diff
@@ -116,7 +116,7 @@ pub struct ChannelHandshakeConfig {
 	/// channels unless [`ChannelHandshakeLimits::force_announced_channel_preference`] is set.
 	///
 	/// Default value: `false`
-	pub announced_channel: bool,
+	pub is_announced_for_forwarding: bool,
```

Comment:
I'd lean more towards `announce_for_forwarding` to align more with the phrasing used in other fields like `negotiate_scid_privacy`. Happy to keep it as is, though, if we are in rush to land this.